.github/workflows/build-and-deploy.yaml

```diff
@@ -0,0 +1,57 @@
+# For more details and explanations see:
+# https://github.com/google-github-actions/setup-gcloud/tree/master/example-workflows/cloud-run
+
+# Copyright 2019 Google, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+on:
+  push:
+    branches:
+      - "main"
+
+name: Build and Deploy to Cloud Run
+env:
+  PROJECT_ID: ${{ secrets.GCP_PROJECT }}
```

Comment:
Why are you storing a secret in `ENV`? it should just be used from the secrets instead of env.